### PR TITLE
Disallow updating properties after deleting their entities

### LINKED ISSUE #1491: Server crashes when attempting to update deleted entity.
Running the query:
```
MATCH (a) DETACH DELETE a SET a.v = 5
```
Causes a server crash with the backtrace:
```
Thread 4 "thread-pool-0" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff5a24700 (LWP 233434)]
0x00007ffff6d2a67e in GraphEntity_GetProperty (e=0x7fffec00ff04, attr_id=9) at /home/tower-linux/dev/redis/graph/src/graph/entities/graph_entity.c:73
73                      if(attr_id == e->entity->properties[i].id) {
(gdb) bt
#0  0x00007ffff6d2a67e in GraphEntity_GetProperty (e=0x7fffec00ff04, attr_id=9) at /home/tower-linux/dev/redis/graph/src/graph/entities/graph_entity.c:73
#1  0x00007ffff6ca8d43 in _UpdateEntity (ge=0x7fffec00ff04, update=0x7fffec00ff04) at /home/tower-linux/dev/redis/graph/src/execution_plan/ops/op_update.c:29
#2  0x00007ffff6ca8ea0 in _UpdateNode (op=0x7fffec007e40, updates=0x7fffec00ff04, update_count=1)
    at /home/tower-linux/dev/redis/graph/src/execution_plan/ops/op_update.c:91
#3  0x00007ffff6ca8f9c in _CommitEntityUpdates (op=0x7fffec007e40, ctx=0x7fffec007f1c)
    at /home/tower-linux/dev/redis/graph/src/execution_plan/ops/op_update.c:119
#4  0x00007ffff6ca9055 in _CommitUpdates (op=0x7fffec007e40) at /home/tower-linux/dev/redis/graph/src/execution_plan/ops/op_update.c:133
#5  0x00007ffff6caa3ad in UpdateConsume (opBase=0x7fffec007e40) at /home/tower-linux/dev/redis/graph/src/execution_plan/ops/op_update.c:341
```

### PR DESCRIPTION
**Pull request by jeffreylovitz**
Resolves #1491

### CODE REVIEW

File: src/graph/entities/graph_entity.c [hunk 1]
```diff
@@ -206,12 +206,17 @@ void GraphEntity_ToString(const GraphEntity *e, char **buffer, size_t *bufferLen
 	*bytesWritten += snprintf(*buffer + *bytesWritten, *bufferLen, "%s", closeSymbole);
 }
 
+inline bool GraphEntity_IsDeleted(const GraphEntity *e) {
+	return DataBlock_ItemIsDeleted(e->entity);
```
Comment (swilly22) [inline src/graph/entities/graph_entity.c:210]: I feel like accessing the `DataBlock` straight from `graph_entity` breaks abstraction to some extent, seems to me like calling `Graph_EntityIsDeleted` is the way to go, (obviously need to introduce `Graph_EntityIsDeleted`).

File: src/execution_plan/ops/op_update.c [hunk 2]
```diff
@@ -25,6 +25,12 @@ static int _UpdateEntity(GraphEntity *ge, PendingUpdateCtx *update) {
 	SIValue new_value = update->new_value;
 	Attribute_ID attr_id = update->attr_id;
 
+	// If this entity has been deleted, perform no updates and return early.
+	if(GraphEntity_IsDeleted(ge)) {
+		res = 0;
+		goto cleanup;
+	}
+
```
Comment (swilly22) [inline src/execution_plan/ops/op_update.c:33]: Would it be cheaper to catch this at `_EvalEntityUpdates` somewhere around line 246, that way we won't be carrying unnecessary updates with us

**Review by swilly22 — CHANGES_REQUESTED** [review]
Overall very nice PR!
Just a few suggestions for improvement

File: src/execution_plan/ops/op_update.c (hunk 2 above)
Comment (jeffreylovitz) [reply src/execution_plan/ops/op_update.c:33]: That would actually be too early; the `DeleteFree` routine needs to execute before entities are properly marked as deleted.

**Review by swilly22 — APPROVED** [review]

**Pull request merged**